hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReader.java

```diff
@@ -497,6 +509,11 @@ public Builder<T> withAllowInflightInstants(boolean allowInflightInstants) {
       return this;
     }
 
+    public Builder<T> withFilter(Option<Predicate> filter) {
```

Comment:
It looks like this is only applied to the keys. If that is the intention then the naming should reflect that